src/geometry/CTiglBSplineAlgorithms.cpp

```diff
@@ -546,12 +547,44 @@ std::vector<Handle(Geom_BSplineSurface) > CTiglBSplineAlgorithms::createCommonKn
     }
 
     // first in u direction
-    makeGeometryCompatibleImpl(adapterSplines, 1e-15);
+    makeGeometryCompatibleImpl(adapterSplines, 1e-14);
 
     for (size_t i = 0; i < old_surfaces_vector.size(); ++i) adapterSplines[i].setDir(vdir);
 
     // now in v direction
-    makeGeometryCompatibleImpl(adapterSplines, 1e-15);
+    makeGeometryCompatibleImpl(adapterSplines, 1e-14);
+
+    return std::vector<Handle(Geom_BSplineSurface)>(adapterSplines.begin(), adapterSplines.end());
+}
+
+std::vector<Handle(Geom_BSplineSurface) > CTiglBSplineAlgorithms::createCommonUKnotsVectorSurface(const std::vector<Handle(Geom_BSplineSurface) >& old_surfaces_vector)
+{
+    // all B-spline surfaces must have the same parameter range in u- and v-direction
+    // TODO: Match parameter range
+
+    // Create a copy that we can modify
+    std::vector<SurfAdapterView> adapterSplines;
+    for (size_t i = 0; i < old_surfaces_vector.size(); ++i) {
+        adapterSplines.push_back(SurfAdapterView(Handle(Geom_BSplineSurface)::DownCast(old_surfaces_vector[i]->Copy()), udir));
+    }
+
+    makeGeometryCompatibleImpl(adapterSplines, 1e-14);
+
+    return std::vector<Handle(Geom_BSplineSurface)>(adapterSplines.begin(), adapterSplines.end());
+}
+
+std::vector<Handle(Geom_BSplineSurface) > CTiglBSplineAlgorithms::createCommonVKnotsVectorSurface(const std::vector<Handle(Geom_BSplineSurface) >& old_surfaces_vector)
+{
+    // all B-spline surfaces must have the same parameter range in u- and v-direction
+    // TODO: Match parameter range
```

Comment:
In principle, you are mentioning two issues here:
  - Code repetition. I'll fix that as proposed
  - The remaining TODO w.r.t match parameter range. The only question here is, what parameter range should be targetet, if the curves to match. Should be reparametrize them into [0,1]?

Reply:
Good question. Why was the To-Do added in the first place? Is it really necessary to match the parameter range when matching the knot vectors? Or would it be error-prone to let the function do two things, which is not  apparent from the function name/signature?

So we could either
 - Remove the To-Do
 - Match the parameters just like it is done in `CTiglBSplineAlgorithms::createCommonKnotsVectorCurve`: Here the parameters are matched to the parameter range of the first spline. In this case, we could also make the parameter range an additional optional argument.

Reply:
> Here the parameters are matched to the parameter range of the first spline. In this case, we could also make the parameter range an additional optional argument.

What I don't like about this approach is the breaking of symmetry. Depending on which argument / geometry is first, the result will be different.

When thinking about it, we should just remove the todo. It should be the responsibilty of the caller to have the parameter ranges matched. Zen if Python: Explicit is better than implicit.

Reply:
> When thinking about it, we should just remove the todo.

I agree. I wonder if in consequence we should remove the reparametrization from `CTiglBSplineAlgorithms::createCommonKnotsVectorCurve`: This would be more consistent and according to your argument this breaks symmetry as well. But who knows what *that* will break... 😨

Reply:
Lets not address too many unrelated things. We could make a new issue from that.

Reply:
I created another issue